SystemManager/Src/SM_States.cpp

```diff
@@ -0,0 +1,243 @@
+    //     system_manager->setState(DisarmMode::getInstance());
+    // }
+
+    // Get waypoint(s) from TM/RC (through LOS) and pass SF data and waypoint(s)
+    // to PM (who sends to AM)
+    // TODO
+
+    // Get latest message back from AM
+    // TODO
+
+    // Create to-ground packet with AM response and SF and send through mail queue to TM
+    // TODO
+}
+
+void FlightMode::exit(SystemManager *system_manager) {
+    // Stop the AM, PM thread and decrease TM speed
+
+    // Stop PM
+    // vTaskDelete(system_manager->PM_handle);
+    // Stop AM
+    vTaskDelete(system_manager->AM_handle);
+
+    // Slow TM operation speed
+    // vTaskDelete(system_manager->TM_handle);
+    // xTaskCreate(system_manager->TMSlowTask, "TM Slow Thread", &system_manager->TM_handle);
+
+    osMessageQueueReset(system_manager->SM_to_AM_queue);
+    system_manager->SM_to_AM_queue = NULL;
+
+    // osMessageQueueReset(system_manager->SM_to_PM_queue);
+    // system_manager->SM_to_PM_queue = NULL;
+    // osMessageQueueReset(system_manager->PM_to_AM_queue);
+    // system_manager->PM_to_AM_queue = NULL;
+
+    // Write 0 to LOS actuators
+    for (uint8_t channel; channel < NUM_ACTUATOR_CHANNELS; channel++) {
+        Los_Actuators::getInstance().set(channel, 0);
+    }
+}
+
+SystemState &FlightMode::getInstance() {
+    static FlightMode singleton;
+    return singleton;
+}
+
+// CommandsFromSM* FlightMode::generatePmPacket(SystemManager *system_manager) {
+//     // Pack SF and Waypoint data to packet sent to PM
+//     CommandsFromSM pm_packet;
+
+//     pm_packet = system_manager->sf_data;
+
+//     if (system_manager->RC_new_data) { // TODO make this match what we get
+//         system_manager->AM_Input = getManualWaypoint(system_manager->RC_msg);
+//     }
+
+//     //TODO
+//     return &pm_packet;
+// }
+
+/********************
+ * FatalFailure Mode
+ ********************/
+
+void FatalFailureMode::enter(SystemManager *system_manager) {
+    // Stop PM
+    // vTaskDelete(system_manager->PM_handle); Not yetrunning for Nov 27th.
+    // Stop AM
+    vTaskDelete(system_manager->AM_handle);
+
+    // For Nov 27th just free the SM to AM queue
+    osMessageQueueReset(system_manager->SM_to_AM_queue);
+    system_manager->SM_to_AM_queue = NULL;
+
+    // osMessageQueueReset(system_manager->SM_to_PM_queue);
+    // system_manager->SM_to_PM_queue = NULL;
+    // osMessageQueueReset(system_manager->PM_to_AM_queue);
+    // system_manager->PM_to_AM_queue = NULL;
+    // osMessageQueueReset(system_manager->PM_to_AM_queue);
+    // system_manager->AM_to_SM_queue = NULL;
+
+    // Write 0 to LOS actuators
+    for (uint8_t channel; channel < NUM_ACTUATOR_CHANNELS; channel++) {
+        Los_Actuators::getInstance().set(channel, 0);
+    }
+}
+
+void FatalFailureMode::execute(SystemManager *system_manager) {
+    // Maybe report something? Log timstamps to SD card?
+    // TBD
+    system_manager->setState(FatalFailureMode::getInstance());
```

Comment:
Do we need to continually setState, or will it just re-run the same state if we don't?

Reply:
I understand that, but I guess I was wondering if there is any need to specify that the next state is FatalFailure if we are already in FatalFailure. I dont think it matters though, we can resolve this.